In [33]:
import pandas as pd
import numpy as np

In [8]:
train = pd.read_csv(r"C:\Users\SRK\Downloads\ipynb\predict_future_sales\sales_train.csv")
items = pd.read_csv(r"C:\Users\SRK\Downloads\ipynb\predict_future_sales\items.csv")
item_cat = pd.read_csv(r"C:\Users\SRK\Downloads\ipynb\predict_future_sales\item_categories.csv")
shops = pd.read_csv(r"C:\Users\SRK\Downloads\ipynb\predict_future_sales\shops.csv")
test = pd.read_csv(r"C:\Users\SRK\Downloads\ipynb\predict_future_sales\test.csv")

In [10]:
count = 0
for i in train['item_cnt_day']:
    if i < -1:
        count = count + 1
count


104

### Highlights:
Some of the records in item sold per day are negative. The record of the item can be equal to zero or greater but can't be less than zero. Hope my intuition works.

In [29]:
print("---------------------------TRAIN-------------------------------")
train.info()
print("---------------------------ITEMS-------------------------------")

items.info()
print("---------------------------ITEM_CAT---------------------------")
item_cat.info()
print("---------------------------SHOPS------------------------------")
shops.info()
print('---------------------------TEST-------------------------------')
test.info()

---------------------------TRAIN-------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2935849 entries, 0 to 2935848
Data columns (total 6 columns):
 #   Column          Dtype  
---  ------          -----  
 0   date            object 
 1   date_block_num  int64  
 2   shop_id         int64  
 3   item_id         int64  
 4   item_price      float64
 5   item_cnt_day    float64
dtypes: float64(2), int64(3), object(1)
memory usage: 134.4+ MB
---------------------------ITEMS-------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22170 entries, 0 to 22169
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   item_name         22170 non-null  object
 1   item_id           22170 non-null  int64 
 2   item_category_id  22170 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 519.7+ KB
---------------------------ITEM_CAT---------------------------
<class 'pan

### Highlights:
The data is recorded from Jan 2013 to Oct 2015 with duration i.e., 32 months. We have 6 variables in total. Every variable can be understood clearly while 'date_block_num' is different. The Jan 2013 is represented with 0, Feb 2013 is represented as 1, Mar 2013 as 2 and so on untill Oct 2015. Our task is to predict sales of Nov 2015.

In [12]:
train['item_cnt_day'].max()

2169.0

In [13]:
train['item_id'].value_counts()

20949    31340
5822      9408
17717     9067
2808      7479
4181      6853
         ...  
593          1
13866        1
16574        1
20668        1
0            1
Name: item_id, Length: 21807, dtype: int64

### Highlights:
In the above cell, the item with id 20949 occured 31340 times. Also we can clearly see that each item is sold atleast once in the given time period.

In [14]:
train.isnull().sum()


date              0
date_block_num    0
shop_id           0
item_id           0
item_price        0
item_cnt_day      0
dtype: int64

### Highights:
It's a good thing that we don't have any missing values. Based on analysis, none of the datasets have missing values.

In [37]:
items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22170 entries, 0 to 22169
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   item_name         22170 non-null  object
 1   item_id           22170 non-null  int64 
 2   item_category_id  22170 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 519.7+ KB


In [38]:
test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 214200 entries, 0 to 214199
Data columns (total 3 columns):
 #   Column   Non-Null Count   Dtype
---  ------   --------------   -----
 0   ID       214200 non-null  int64
 1   shop_id  214200 non-null  int64
 2   item_id  214200 non-null  int64
dtypes: int64(3)
memory usage: 4.9 MB


### Highlights:
There are 22169 items in total. 

In [16]:
test.isnull().sum()

ID         0
shop_id    0
item_id    0
dtype: int64

### Highlights:
We have test data which is 7% of train data.

In [31]:
print('Number of duplicates:', len(train[train.duplicated()]))

Number of duplicates: 6


In [41]:
def downcast_dtypes(train):
    float_cols = [c for c in train if train[c].dtype == "float64"]
    int_cols = [c for c in train if train[c].dtype in ["int64", "int32"]]
    train[float_cols] = train[float_cols].astype(np.float32)
    train[int_cols] = train[int_cols].astype(np.int16)
    return train

sale_train = downcast_dtypes(train)
print(train.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2935849 entries, 0 to 2935848
Data columns (total 6 columns):
 #   Column          Dtype  
---  ------          -----  
 0   date            object 
 1   date_block_num  int16  
 2   shop_id         int16  
 3   item_id         int16  
 4   item_price      float32
 5   item_cnt_day    float32
dtypes: float32(2), int16(3), object(1)
memory usage: 61.6+ MB
None


####  DOWN CASTING:
Downcasting means we reduce the datatypes of each feature to its lowest possible type.Here we converted int32 to int16, float64 to float32 and many.Also the memory usage is reduced from 134MB to 61MB Such a huge data must be casted to carryout the techniques involved in pre-processing.



